File: lintcode/Find_Minimum_in_Rotated_Sorted_Array.cpp
class Solution {
public:
    /**
     * @param num: a rotated sorted array
     * @return: the minimum number in the array
     */
    int findMin(vector<int> &num) {
        // write your code here
        if(num.size() == 0) throw runtime_error("empty input!");
        
        int l = 0;
        int h = num.size()-1;
        while(l <= h){
            if(l == h) return num[l];
            int m = l + (h-l)/2;
            if(num[m] < num[h]) h = m;
            else l = m+1;
        }
        
        throw runtime_error("should not be here!");
    }
};

File: lintcode/java/FindMinimuminRotatedSortedArray.java
public class Solution {
    /**
     * @param num: a rotated sorted array
     * @return: the minimum number in the array
     */
    public int findMin(int[] A) {
        if(A.length <= 0) throw new IllegalArgumentException("xxx");
        
        int l = 0;
        int h = A.length-1;
        int m;
        
        while(l <= h){
            if(l == h) {
                return A[l];
            }
            m = l + (h-l)/2;
            if(A[m] < A[h]){
                h = m;
            } else {
                l = m+1;
            }
        }
        
        throw new RuntimeException("should not be there");
    }
}
